### PR TITLE
complain if _locales specified but no default_locale

### LINKED ISSUE #701: Validate `default_locale` is being set if _locales/ folder exists
Supports https://github.com/mozilla/addons-server/issues/1681

From https://developer.chrome.com/extensions/i18n:

> Important: If an extension has a _locales directory, the manifest must define "default_locale"

### PR DESCRIPTION
**Pull request by andymckay**
* fixes #701 which is kind of the inverse of the last one.
* not sure how to do this without the loop, but it should be pretty damn fast on anything but the biggest add-ons

### CODE REVIEW

File: src/parsers/manifestjson.js [hunk 1]
```diff
@@ -140,6 +140,16 @@ export default class ManifestJSONParser extends JSONParser {
         this.isValid = false;
       }
     }
+
+    if (!this.parsedJSON.default_locale && this.io) {
+      for (let key in this.io.files) {
```
Comment (tofumatt) [inline src/parsers/manifestjson.js]: `filePath` would be a more appropriate variable name.

File: src/parsers/manifestjson.js [hunk 2]
```diff
@@ -140,6 +140,16 @@ export default class ManifestJSONParser extends JSONParser {
...
+        if (key.startsWith('_locales')) {
```
Comment (tofumatt) [inline src/parsers/manifestjson.js]: Like you said I'm not a huge fan of the loop but our IO class doesn't yet have a usable `checkIfPathExists` type method. I've filed #1079 for it, though it's obviously low-priority.

**Comment by andymckay** [comment]
I'm not sure where AMO will have a problem, but @EnTeQuAk might no more. Let's assume that having a `messages.json` inside `_locales` is the problem, but not an empty `_locales`. I've updated it to be a regex and look for `messages.json`. Because its a regex added more tests. How does that look?

File: src/messages/manifestjson.js [hunk 3]
```diff
@@ -117,3 +117,13 @@ export const NO_MESSAGES_FILE = {
     exists. See: https://mzl.la/2hjcaEE`),
   file: MANIFEST_JSON,
 };
+
+export const NO_DEFAULT_LOCALE = {
+  code: 'NO_DEFAULT_LOCALE',
+  legacyCode: null,
+  message: _('The "default_locale" is missing but "_locales" exist.'),
+  description: _(singleLineString`The "default_locale" value is not specifed in
+    the manifest, but a "_locales" directory exists.
```
Comment (tofumatt) [inline src/messages/manifestjson.js]: I would say `a "locales" directory with "messages.json" files`, just so they know exactly what to look for.

**Comment by andymckay** [comment]
It wont catch `_locales/messages.json` because there has to be two slashes in there so `_locales//messages.json` would fail, I added a test for that. But lets face it, if we've got files like that, things are messed up somewhere.

**Comment by tofumatt** [comment]
Agreed.

Cool, looks good to me! :-)

**Pull request merged**